docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+BTR generates a screenshot for each of the paths rendered during the build in the output/info/screenshots directory of your project.
+
+### Hash history
+
+Build time rendering supports applications that use either the `@dojo/framework/routing/history/HashHistory` or `@dojo/framework/routing/history/StateHistory` history managers. If your application uses the HashHistory, ensure that all paths are prefixed with a # character.
+
+## build-time-render feature
+
+Build time rendering exposes a `build-time-render` feature flag that can be used to skip functionality that cannot be executed at build time. This can be used to avoid making `fetch` calls to external systems and instead provide static data that can be used to create an initial render.
+
+```ts
+if (has('build-time-render')) {
+	const response = await fetch(/* remote JSON */);
+	return response.json();
+} else {
+	return Promise.resolve({
+		/* predefined Object */
+	});
+}
+```
+
+## Blocks meta
+
+Dojo Blocks allow executing code in Node.js as part of the build time rendering process. The results from the execution are written to a cache that can then be transparently used in the same way at runtime in the browser. This opens up new opportunities for performing operations that might be not possible or unperformant in a browser.
+
+For example, a Dojo Block module could read a group of markdown files, transform them into VNodes, and make them available to render in the application, all at build time. The result of this Dojo Block module is then cached into the application bundle for use at runtime in the browser.
+
+A Dojo Block module gets used like any other meta in any Dojo widget. As a result there is no extensive configuration or alternative authoring patterns needed.
+
+For example, a block module could read a text file and return the content to the application.
+
+> src/read-file.block.ts
+
+```ts
+import * as fs from 'fs';
+import { resolve } from 'path';
+
+export default (path: string) => {
+	path = resolve(__dirname, path);
+	return fs.readFileSync(path, 'utf8');
+};
+```
+
+> src/MyBlockWidget.ts
+
+```ts
+import Block from '@dojo/framework/widget-core/meta/Block';
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { v } from '@dojo/framework/widget-core/d';
+
+import readFile from './read-file.block';
+
+export default class MyBlockWidget extends WidgetBase {
+	protected render() {
+		const message = this.meta(Block).run(readFile)('../content/hello-dojo-blocks.txt');
+		return v('div', [message]);
+	}
+}
+```
+
+This widget runs the `read-file.block.ts` module at build time, loading the file path passed which gets used as the children in the widget.
+
+## Snapshots
```

Comment:
This section does not feel necessary... the result of build time rendering should be explain in the main section?

Reply:
I think this came a conversation I had about when the BTR was captured. I remember working with SSR solutions and deciding when to capture the DOM was always an issue.